Document new env variable in Qiskit 1.2.0 (#1836)
Qiskit/documentation: docs/guides/configure-qiskit-local.mdx

- `QISKIT_NUM_PROCS`: The maximum number of parallel processes to launch for parallel operations if parallel execution is enabled. Specify an integer greater than zero.
- `RAYON_NUM_THREADS`: The number of threads to run multithreaded operations in Qiskit. By default, multithreaded code launches a thread for each logical CPU. To adjust the number of threads Qiskit uses, set this to an integer value. For example, setting RAYON_NUM_THREADS=4 launches four threads for multithreaded functions. 
- `QISKIT_FORCE_THREADS`: Specifies that multithreaded code should always execute in multiple threads. By default, if you're running multithreaded code in a section of Qiskit that is already running in parallel processes, Qiskit does not launch multiple threads but instead executes that function serially. This is done to avoid potentially overloading limited CPU resources. However, if you want to force the use of multiple threads even when in a multiprocess context, set `QISKIT_FORCE_THREADS=TRUE`.

> - `QISKIT_SABRE_ALL_THREADS`: Is used to control the behavior of the layout and routing pass in Qiskit's preset pass manager. When set to `1` or `TRUE` this will use all available CPUs for running multiple random trials. This can improve the quality of the results, especially for systems with > 20 CPUs/cores, but comes with the tradeoff of results not being reproducible when run on different local hardware.

Reviewer: ```suggestion
- `QISKIT_SABRE_ALL_THREADS`: Is used to control the behavior of the layout and routing pass in Qiskit's preset pass manager. When set to `1` or `TRUE`, this will use all available CPUs for running multiple random trials. This can improve the quality of the results, especially for systems with > 20 CPUs/cores, but comes with the tradeoff of results not being reproducible when run on different local hardware.
```

> - `QISKIT_SABRE_ALL_THREADS`: Is used to control the behavior of the layout and routing pass in Qiskit's preset pass manager. When set to `1` or `TRUE`, this will use all available CPUs for running multiple random trials. This can improve the quality of the results, especially for systems with > 20 CPUs/cores, but comes with the tradeoff of results not being reproducible when run on different local hardware.

Reviewer 2: ```suggestion
- `QISKIT_SABRE_ALL_THREADS`: Controls the behavior of the layout and routing pass in Qiskit's preset pass manager. When set to `1` or `TRUE`, this uses all available CPUs for running multiple random trials. This can improve the quality of the results, especially for systems with greater than 20 CPUs/cores; the tradeoff, however, is that results are not reproducible when run on different local hardware.
```
